net/raft/raft.go

```diff
@@ -735,10 +745,10 @@ func (sv *Service) applyEntry(ent raftpb.Entry, writers map[string]chan bool) {
 		case raftpb.ConfChangeAddNode, raftpb.ConfChangeUpdateNode:
 			sv.state.SetPeerAddr(cc.NodeID, string(cc.Context))
 		case raftpb.ConfChangeRemoveNode:
+			sv.state.RemovePeerAddr(cc.NodeID)
 			if cc.NodeID == sv.id {
-				panic(errors.New("removed from cluster"))
+				os.Exit(0)
```

Comment:
This was basically for testing: `panic` would cause the test to crash with an error and I couldn't `recover` from the test code since the `panic` was in a different goroutine. Also, since this is expected behavior when a node is removed from the cluster, I'm not sure if a panic makes more sense than just exiting.